how-to/security/openssh-server.md

```diff
@@ -101,226 +103,12 @@ ssh-import-id <username-on-remote-service>
-
-* Use the 2FA device's backup or cloud sync facility if it has one.
-* Write down the backup codes printed by the setup tool.
-* Take a photo of the QR code.
-* (TOTP only) Scan the QR code on multiple 2FA devices. This only works for TOTP, since multiple HOTP 2FA devices will not be able to stay in sync.
-* Ensure that the user has a different authentication path to be able to rerun the setup tool if required.
-
-Of course, any of these backup steps also negate any benefit of 2FA should someone else get access to the backup, so the steps taken to protect any backup should be considered carefully.
-
-### Configure the SSH server
-
-Once all users are configured, configure `sshd` itself by editing `/etc/ssh/sshd_config`. Depending on your installation, some of these settings may be configured already, but not necessarily with the values required for this configuration. Check for and adjust existing occurrences of these configuration directives, or add new ones, as required:
-
-```
-KbdInteractiveAuthentication yes
-PasswordAuthentication no
-AuthenticationMethods publickey,keyboard-interactive
-```
-
-> **Note**:
-> On Ubuntu 20.04 "Focal Fossa" and earlier, use `ChallengeResponseAuthentication yes` instead of `KbdInteractiveAUthentication yes`.
-
-Restart the `ssh` service to pick up configuration changes:
-
-```bash
-sudo systemctl try-reload-or-restart ssh
-```
-
-Edit `/etc/pam.d/sshd` and replace the line:
-
-```
-@include common-auth
-```
-
-with:
-
-```
-auth required pam_google_authenticator.so
-```
-
-Changes to PAM configuration have immediate effect, and no separate reloading command is required.
-
-### Log in using 2FA
-
-Now when you log in using SSH, in addition to the normal public key authentication, you will be prompted for your TOTP or HOTP code:
-
-```bash
-$ ssh jammy.server
-Enter passphrase for key 'id_rsa':
-(ubuntu@jammy.server) Verification code:
-Welcome to Ubuntu Jammy Jellyfish...
-(...)
-ubuntu@jammy.server:~$
-```
-
-### Special cases
-
-On Ubuntu, the following settings are default in `/etc/ssh/sshd_config`, but if you have overridden them, note that they are required for this configuration to work correctly and must be restored as follows:
-
-```
-UsePAM yes
-PubkeyAuthentication yes
-```
+## Two factor authentication
 
-Remember to run `sudo systemctl try-reload-or-restart ssh` for any changes made to `sshd` configuration to take effect.
+You can add an extra layer of security to the default key-based authentication using two factor authentication. You can add two factor authentication {ref}`using U2F/FIDO hardware authentication devices <two-factor-authentication-with-u2f-or-fido.md>`. Alternatively, in cases U2F/FIDO hardware authentication devices are unavailable or impractical for your use case you can add it {ref}`using HMAC/Time based One Time Passwords (HOTP/TOTP) <two-factor-authentication-with-TOTP-or-HOTP.md>`.
```

Comment:
```suggestion
You can add an extra layer of security to the default key-based authentication using two factor authentication. You can add two factor authentication {ref}`using U2F/FIDO hardware authentication devices <two-factor-authentication-with-u2f-or-fido>`. Alternatively, in cases U2F/FIDO hardware authentication devices are unavailable or impractical for your use case you can add it {ref}`using HMAC/Time based One Time Passwords (HOTP/TOTP) <two-factor-authentication-with-TOTP-or-HOTP>`.
```

Reply:
The links are not showing up in the preview build - when using the {ref} role, it's pointing to the anchor inside the file (rather than the file itself), so including the file extension here is probably what's breaking it.